Test monitor notifications (#3977)
DataDog/documentation: content/monitors/notifications.md

{{% /tab %}}
{{< /tabs >}}

## Test monitor notifications

After you define your monitor, you can test what your monitor's notification would look like in any applicable state. You can test what's specified in the "Say What's Happening" field by selecting "Select all" or selecting specific states.

{{< img src="monitors/notifications/test-notif-select.png" alt="Test the notifications for this monitor" responsive="true" style="width:50%;" >}}

You can only test states that are available in the monitor’s configuration and test for thresholds specified in the alerting conditions. Recovery thresholds are an exception, as Datadog sends a recovery notification once the monitor is no longer in alert or has no warn conditions.

> 

Reviewer: Can we link recovery thresholds to this: https://docs.datadoghq.com/monitors/faq/what-are-recovery-thresholds/ 

This way a user can continue reading about recovery thresholds since this input is optional

Author: 👍

In the following, you are only able to test for ALERT because only an alert threshold has been specified:

> {{< img src="monitors/notifications/test-notif-alert.png" alt="Set alert conditions" responsive="true" style="width:50%;" >}}

Reviewer 2: This image is small and a little fuzzy. I would replace.

In this next example, you can test for both ALERT and NO DATA:

> {{< img src="monitors/notifications/test-notif-alert-nodata.png" alt="Test the notifications for this monitor" responsive="true" style="width:50%;" >}}

Reviewer 2: Same with this image

Message variables auto-populate with an available randomly selected group based on the scope of your monitor's definition.

{{< img src="monitors/notifications/test-notif-message.png" alt="Say what's happening" responsive="true" style="width:50%;" >}}

### Managing test notifications

> Test notifications produce events that can be searched for within the event stream. These notification walso indicate who initiated the test in the message body, and [TEST] is added into the test notification title. 

Reviewer 2: ```suggestion
Test notifications produce events that can be searched for within the event stream. These notifications also indicate who initiated the test in the message body, and [TEST] is added into the test notification title. 
```

or 

```suggestion
Test notifications produce events that can be searched for within the event stream, indicate who initiated the test in the message body, and include [TEST] in the notification title. 
```

Your test notifications are sent to any notification handle available in the message box. You can also test webhooks.  

**TIP**: Datadog recommends that you test using a test notification channel if you do not want to add these notifications to your on-call channels. 

> You do not need to save the monitor to run the notification test. By clicking Run Test, the notification is sent, and you can continue to edit the monitor and notification box if you wish before saving. 

Reviewer 2: ```suggestion
You do not need to save the monitor to run the notification test. Clicking **Run test** sends the notification and you can continue to edit the monitor and notification box before saving. 
```

> Testing notifications are currently supported for the following monitor types: Host, Metric, Anomaly, Outlier, Forecast, Integration (check only), Process (check only), Network (check only), Custom check, and Event

Reviewer 2: ```suggestion
Testing notifications are currently supported for the following monitor types: Host, Metric, Anomaly, Outlier, Forecast, Integration (check only), Process (check only), Network (check only), Custom check, and Event.
```
